contracts/finality/Cargo.toml

```diff
@@ -1,5 +1,5 @@
 [package]
-name = "op-finality-gadget"
+name = "finality"
```

Comment:
I agree this can be called just `finality`, as the reference / generic impl. Other contracts can live in this repo in the future as well, and have more specific names.

Also, if we move common functionality to packages, these contracts can be like thin layers on top of common functions, in which implementors chose / adapt the functionality to their own needs.